numpy/distutils/command/build_ext.py

```diff
@@ -570,12 +570,13 @@ def _process_unlinkable_fobjects(self, objects, libraries,
         unlinkable_fobjects = list(unlinkable_fobjects)
 
         # Expand possible fake static libraries to objects
+        to_remove = []
         for lib in libraries:
```

Comment:
If we know that libraries is a list, could also do `libraries[:]`, but not sure that we do, so this is probably good. Should it even mutate the libraries that were passed in?

Reply:
The list does need to change or else the linker will try to link nonexistent or empty `.lib` files and crash.  `libraries` is converted to a list at the top of this function

Reply:
`libraries[:]` will mean it iterates a copy of the list, which should be safe, basically undoing the PR you linked.

Reply:
Changed to `libraries[:]` in latest push

Reply:
Would it be possible to create a short regression test for this (I am just fishing here, but every test helps).

Reply:
A simple build test (exactly like the one linked in description) could do the job.  I'm not familiar with the numpy conventions of how to do this.  What I have in mind is building an extension and then testing to see if the extension exists (i.e., was built successfully).  Is there a simple example you can point me to?

Reply:
Test merged